socket.go

```diff
@@ -30,12 +31,13 @@ var (
 
 // socket implements the ZeroMQ socket interface
 type socket struct {
-	ep    string // socket end-point
-	typ   SocketType
-	id    SocketIdentity
-	retry time.Duration
-	sec   Security
-	log   *log.Logger
+	ep         string // socket end-point
+	typ        SocketType
+	id         SocketIdentity
+	retry      time.Duration
+	sec        Security
+	log        *log.Logger
+	maxRetries int
```

Comment:
perhaps put it closer to `retry`?
(they are logically related)

Reply:
thx for the review, I moved it up